### ISSUE TITLE
[CP] [Impeller] Fix stroke cap drawing not correct

### ISSUE THREAD

**Issue by luckysmg** [flutter/flutter#124176]
### issue_link

https://github.com/flutter/flutter/issues/119810

### Commit Hash

[24fd06e](https://github.com/flutter/engine/commit/24fd06e79767e4f9bbe69bf15390760756b23ae2)

### Target

stable

### pr_link

https://github.com/flutter/engine/pull/39481

### Impacted Users

For user using impeller to draw path

### Impact Description

Impact when user uses impeller to draw stroke path. Without this patch they will see a one-side cap, which is not expected
<img src="https://user-images.githubusercontent.com/49340347/229960495-3b971c46-95d4-4ca0-b1e7-47b1b692ac99.png" width=300 />

### Workaround

Seems no workaround.

### Risk

low

### Test Coverage

yes

### Validation Steps

Using impeller to draw a stroke path, and set the cap to `StrokeCap.round`,  expected to see both side cap rendering is good.

```dart
void main() {
  runApp(CustomPaint(
    painter: MyPainter(),
  ));
}

class MyPainter extends CustomPainter {
  @override
  void paint(Canvas canvas, Size size) {
    Paint paint = Paint()
      ..color = Colors.red
      ..strokeWidth = 20
      ..strokeCap = StrokeCap.round;
    canvas.drawLine(const Offset(100, 100), const Offset(200, 100), paint);
  }

  @override
  bool shouldRepaint(covariant CustomPainter oldDelegate) {
    return true;
  }
}
```

**Comment by luckysmg** [comment]
cc @zanderso @bdero

**Comment by itsjustkevin** [comment]
Friendly ping for @zanderso, @bdero or @jonahwilliams to take a look at this cherry-pick.

**Comment by zanderso** [comment]
This was a good fix, but since it is late in the cycle, and the fix is available in betas 3.8 and 3.9 (and soon 3.10), the risk/benefit speaks against accepting this CP against 3.7, even though the risk is likely very small. So I would say not lgtm, but big thanks to @luckysmg for spotting this and fixing it. Our next stable will be very polished!

**Comment by itsjustkevin** [comment]
Thanks for the feedback @zanderso and thank you for the cherry-pick request @luckysmg, closing this cherry-pick request.
